components/net/filemanager_thread.rs

```diff
@@ -231,8 +238,13 @@ impl<UI: 'static + UIProvider> FileManager<UI> {
 
     fn select_files(&mut self, patterns: Vec<FilterPattern>,
                     sender: IpcSender<FileManagerResult<Vec<SelectedFile>>>,
-                    origin: FileOrigin) {
-        match self.ui.open_file_dialog_multi("", patterns) {
+                    origin: FileOrigin, opt_test_paths: Option<Vec<String>>) {
+        let opt_v = match opt_test_paths {
+            Some(v) => Some(v),
+            None => self.ui.open_file_dialog_multi("", patterns),
+        };
```

Comment:
This isn't process safe -- we should assert that the pref is enabled if opt_test_paths is nonempty here.